larpandoracontent/LArThreeDReco/LArCosmicRay/CosmicRayVertexBuildingAlgorithm.cc

```diff
@@ -172,43 +173,77 @@ void CosmicRayVertexBuildingAlgorithm::BuildCosmicRayParent(const LArPointingClu
                 maxDirection = pointingCluster.GetOuterVertex().GetDirection();
             }
             else
-            {
-                LArClusterHelper::GetExtremalCoordinates(pCluster, minPosition, maxPosition);
+	    {
+	        LArClusterHelper::GetExtremalCoordinates(pCluster, minPosition, maxPosition);
                 minDirection = (maxPosition - minPosition).GetUnitVector();
                 maxDirection = (minPosition - maxPosition).GetUnitVector();
-            }
-
+	    }
+	    
             if ((maxPosition - minPosition).GetMagnitudeSquared() < std::numeric_limits<float>::epsilon())
-                throw StatusCodeException(STATUS_CODE_NOT_FOUND);
-
-            if (!foundVtx || (minPosition.GetY() > std::max(maxPosition.GetY(), vtxPosition.GetY())))
-            {
-                foundVtx = true;
-                vtxPosition = minPosition;
-                vtxDirection = minDirection;
-            }
+   	        throw StatusCodeException(STATUS_CODE_NOT_FOUND);
+	    
+	    if (m_isDualPhase) //X is the vertical direction in Dual-Phase geometry
+	    {
+		if (!foundVtx || (minPosition.GetX() > std::max(maxPosition.GetX(), vtxPosition.GetX())))
```

Comment:
This implementation looks sound to me. The only change I wondered about (and feel free to to dismiss this) might be to cache the actual coordinates used in the comparisons upfront, storing y values by default and x values if in dual phase mode. The logic itself then doesn't need to be repeated, and the choice of using y or x as inputs to the main logic is clarified a little, upfront.

e.g.

```
const float minCoordinate(m_isDualPhase ? minPosition.GetX() : minPosition.GetY());
const float maxCoordinate(m_isDualPhase ? maxPosition.GetX() : maxPosition.GetY());

if (minCoordinate < maxCoordinate) // or some other, similar logic
    outputChosenVector = minPosition;
```